### PR TITLE
CI: Add 3.12 builds

### PR DESCRIPTION
**Pull request by phofl**
Some CoW warnings broke for 3.11, so we should test at least CoW for 3.12 as well

### CODE REVIEW

File: ci/deps/actions-312.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  - numexpr>=2.8.4
+  - odfpy>=1.4.1
+  - qtpy>=2.3.0
+  - pyqt>=5.15.9
+  - openpyxl>=3.1.0
+  - psycopg2>=2.9.6
+  - pyarrow>=10.0.1
+  - pymysql>=1.0.2
+  - pyreadstat>=1.2.0
+  # ∆- pytables>=3.8.0
```
Comment (mroeschke) [inline ci/deps/actions-312.yaml:48]:
```suggestion
  # - pytables>=3.8.0
```

File: pandas/tests/io/excel/test_readers.py [hunk 2]
```diff
@@ -1437,7 +1437,9 @@ def test_deprecate_bytes_input(self, engine, read_ext):
             "byte string, wrap it in a `BytesIO` object."
         )
 
-        with tm.assert_produces_warning(FutureWarning, match=msg):
+        with tm.assert_produces_warning(
+            FutureWarning, match=msg, raise_on_extra_warnings=False
```
Comment (mroeschke) [inline pandas/tests/io/excel/test_readers.py:1441]: What's the extra warning here?

File: pandas/tests/io/excel/test_writers.py [hunk 3]
```diff
@@ -287,7 +287,9 @@ def test_read_excel_parse_dates(self, ext):
 
             date_parser = lambda x: datetime.strptime(x, "%m/%d/%Y")
             with tm.assert_produces_warning(
-                FutureWarning, match="use 'date_format' instead"
+                FutureWarning,
+                match="use 'date_format' instead",
+                raise_on_extra_warnings=False,
```
Comment (mroeschke) [inline pandas/tests/io/excel/test_writers.py:292]: What's the extra warning here?

File: pandas/tests/io/excel/test_readers.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/io/excel/test_readers.py:1441]: a DeprecationWarning that originates in the excel libraries

File: pandas/tests/io/excel/test_writers.py (hunk 3 above)
Comment (phofl) [reply pandas/tests/io/excel/test_writers.py:292]: same as above

**Comment by phofl** [comment]
Yep done, that should do the trick I think

**Review by lithomas1 — APPROVED** [review]
LGTM.

**Pull request merged**